marketplace/config/settings/configurator.py

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""
+Handler module for gathering configuration data.
+"""
+from .env import ENVIRONMENT
+
+
+CLOWDER_ENABLED = ENVIRONMENT.bool("CLOWDER_ENABLED", default=False)
+if CLOWDER_ENABLED:
+    from app_common_python import LoadedConfig, KafkaTopics, ObjectBuckets, SmartAppConfig
+
+class Configurator:
+    """Obtain configuration based on mode."""
+
+    @staticmethod
+    def get_kafka_broker_host():
+        """Obtain kafka broker host address."""
+        if CLOWDER_ENABLED:
+            value = LoadedConfig.kafka.brokers[0].hostname
+        else:
+            value = ENVIRONMENT.get_value("INSIGHTS_KAFKA_HOST", default="localhost")
+        return value
+
+    @staticmethod
+    def get_kafka_broker_port():
+        """Obtain kafka broker port."""
+        if CLOWDER_ENABLED:
+            value = LoadedConfig.kafka.brokers[0].port
+        else:
+            value = ENVIRONMENT.get_value("INSIGHTS_KAFKA_PORT", default="29092")
+        return value
+
+    @staticmethod
+    def get_kafka_topic():
+        """Obtain kafka topic."""
+        if CLOWDER_ENABLED:
+            value = KafkaTopics.get("platform.upload.mkt", "platform.upload.mkt")
```

Comment:
I think this actually returns an object.  Should probably be something like 

```suggestion
            value = KafkaTopics.get("platform.upload.mkt").name
```

You should be able to safely assume the topic will exist since you list it as a requirement in your app spec.  If for some reason Clowder cannot create the topic, it will refuse to deploy the app until the topic is available.

Reply:
ok. updated.